lib/engine/step/train.rb

```diff
@@ -135,10 +135,9 @@ def buyable_trains(entity)
       end
 
       def buyable_train_variants(train, entity)
-        return [] unless buyable_trains(entity).include?(train)
+        return [] unless train_among_buyable?(entity, train)
```

Comment:
this can be simplified

buyable_trains(entity).any? { |train| train.variants[train.name] }

Reply:
Fixed